manager/controllers/app/plotter_generator.go

```diff
@@ -201,6 +201,61 @@ func (p *PlotterGenerator) addStep(element *ResolvedEdge, datasetID string, api
 	return steps
 }
 
+// Handle a new asset: allocate storage and update its metadata. Used when the
+// IsNewDataSet flag is true.
+func (p *PlotterGenerator) HandleNewAsset(item *DataInfo, application *app.FybrikApplication, selection *Solution,
+	plotterSpec *app.PlotterSpec) error {
+	var err error
+	if item.DataDetails != nil && item.DataDetails.Details.DataFormat != "" {
+		return nil
+	}
+	p.Log.Trace().Str(logging.DATASETID, item.Context.DataSetID).Msg("Handle new dataset")
+	var sinkDataStore *app.DataStore
+	var element *ResolvedEdge
+
+	needToAllocateStorage := false
+	for _, element = range selection.DataPath {
+		if element.StorageAccount.Region != "" {
+			needToAllocateStorage = true
+			break
+		}
+	}
+	if !needToAllocateStorage {
+		return nil
+	}
+
+	// allocate storage
+	if sinkDataStore, err = p.AllocateStorage(item, element.Sink.Connection, &element.StorageAccount); err != nil {
+		p.Log.Error().Err(err).Str(logging.DATASETID, item.Context.DataSetID).Msg("Storage allocation failed")
+		return err
+	}
+
+	// Reset StorageAccount to prevent re-allocation
+	element.StorageAccount.Region = ""
+
+	resourceMetadata := datacatalog.ResourceMetadata{
+		Name:      item.Context.DataSetID,
+		Geography: string(element.StorageAccount.Region),
```

Comment:
You have emptied this value in line 234.

Reply:
Fixed. Thanks